imports/wasi_snapshot_preview1/fs.go

```diff
@@ -296,18 +287,21 @@ func fdFdstatSetFlagsFn(_ context.Context, mod api.Module, params []uint64) sysc
 
 	// Currently we only support non-blocking mode for standard I/O streams.
 	// Non-blocking mode is rarely supported for regular files, and we don't
-	// yet have support for sockets so we make a special case.
-	f, ok := fsc.LookupFile(fd)
-	if ok {
-		nonblock := wasip1.FD_NONBLOCK&wasiFlag != 0
-		switch file := f.File.File().(type) {
-		case *sys.StdioFileReader:
-			return platform.UnwrapOSError(file.SetNonblock(nonblock))
-		case *sys.StdioFileWriter:
-			return platform.UnwrapOSError(file.SetNonblock(nonblock))
+	// yet have support for sockets, so we make a special case.
+	if fd <= sys.FdStderr {
```

Comment:
special casing is a little weird, and isn't easy to contain (a couple docs), but keeping the end logic the same helps keep this PR from becoming too gnarly.

Reply:
Does it break if stdin/stdout/stderr are closed and the file descriptor resused?

It was one of the reasons I checked for the internal `sys.StdioFile*` types (my bad for not leaving a comment), tho I am also not 100% sure it was solid.

Reply:
rejigged. thanks!